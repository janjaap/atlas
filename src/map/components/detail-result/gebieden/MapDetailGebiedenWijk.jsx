import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultWrapper from '../MapDetailResultWrapper';
import MapDetailResultItem from '../MapDetailResultItem';

const MapDetailGebiedenWijk = ({ panoUrl, wijk, onMaximize, onPanoPreviewClick }) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={wijk.label}
    title="Wijk"
  >
    <ul className="map-detail-result__list">
      <MapDetailResultItem
        label="Volledige code"
        value={wijk.volledigeCode}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailGebiedenWijk.propTypes = {
  wijk: PropTypes.shape({
    label: PropTypes.string.isRequired,
    volledigeCode: PropTypes.string
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailGebiedenWijk;
