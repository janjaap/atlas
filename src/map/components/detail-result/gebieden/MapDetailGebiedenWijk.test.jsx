import React from 'react';
import { shallow } from 'enzyme';

import MapDetailGebiedenWijk from './MapDetailGebiedenWijk';

describe('MapDetailGebiedenWijk', () => {
  it('should render everything', () => {
    const wijk = {
      label: 'Wijk label',
      volledigeCode: 'A'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailGebiedenWijk
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        wijk={wijk}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
