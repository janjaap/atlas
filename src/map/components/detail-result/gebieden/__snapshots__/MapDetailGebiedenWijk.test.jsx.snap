// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailGebiedenWijk should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Wijk label"
  title="Wijk"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Volledige code"
      value="A"
    />
  </ul>
</MapDetailResultWrapper>
`;
